pkg/apis/maistra/conversion/kiali.go

```diff
@@ -59,6 +60,81 @@ func populateKialiAddonValues(kiali *v2.KialiAddonConfig, values map[string]inte
 		}
 	}
 
+	deploymentConfig := kiali.Install.Deployment
+	if deploymentConfig != nil {
+		resources := deploymentConfig.Resources
+		if resources != nil {
+			if values, err := toValues(resources); err == nil {
+				if len(values) > 0 {
+					if err := setHelmValue(kialiValues, "deployment.resources", values); err != nil {
+						return err
+					}
+				}
+			}
+		}
+
+		affinity := deploymentConfig.Affinity
+		if affinity != nil {
+			nodeAffinity := affinity.NodeAffinity
+			if nodeAffinity != nil {
+				if values, err := toValues(nodeAffinity); err == nil {
+					if len(values) > 0 {
+						if err := setHelmValue(kialiValues, "deployment.affinity.node", values); err != nil {
+							return err
+						}
+					}
+				}
+			}
+
+			podAffinity := affinity.PodAffinity
+			if podAffinity != nil {
+				if values, err := toValues(podAffinity); err == nil {
+					if len(values) > 0 {
+						if err := setHelmValue(kialiValues, "deployment.affinity.pod", values); err != nil {
+							return err
+						}
+					}
+				}
+			}
+
+			podAntiAffinity := affinity.PodAntiAffinity
+			if podAntiAffinity != nil {
+				if values, err := toValues(podAntiAffinity); err == nil {
+					if len(values) > 0 {
+						if err := setHelmValue(kialiValues, "deployment.affinity.pod_anti", values); err != nil {
+							return err
+						}
+					}
+				}
+			}
+		}
+
+		nodeSelector := deploymentConfig.NodeSelector
+		if nodeSelector != nil {
+			if values, err := toValues(nodeSelector); err == nil {
+				if len(values) > 0 {
+					if err := setHelmValue(kialiValues, "deployment.nodeSelector", values); err != nil {
```

Comment:
@luksa should this not be node_selector?  See


      {% if kiali_vars.deployment.node_selector|length > 0 %}
            nodeSelector:
                  {{ kiali_vars.deployment.node_selector | to_nice_yaml(indent=0) | trim | indent(8) }}
      {% else %}
            nodeSelector: null
      {% endif %}

Reply:
never mind, I see we have used nodeSelector in our template but are setting node_selector

Reply:
Yeah, unfortunately, Kiali uses the `node_selector` notation for its values, whereas SMCP uses `nodeSelector`. It was hard for me to decide where to perform the switch, but the existing code in `conversion/kiali.go` already used the latter format, so I kept things consistent.